IbisHardwareIGSIO/plusserverinterface.cpp

```diff
@@ -62,27 +68,37 @@ bool PlusServerInterface::StartServer(const QString& configFilePath)
     // Un-requested log entries that are captured from the PlusServer executable are parsed and dropped from output
     QString cmdLine = QString("\"%1\" --config-file=\"%2\" --verbose=%3").arg(m_plusServerExecutable).arg(configFilePath).arg(m_serverLogLevel);
     m_CurrentServerInstance->start(cmdLine);
-    m_CurrentServerInstance->waitForFinished(500);
+    m_CurrentServerInstance->waitForStarted();
 
     // During waitForFinished an error signal may be emitted, which may delete m_CurrentServerInstance,
     // therefore we need to check if m_CurrentServerInstance is still not NULL
-    if (m_CurrentServerInstance && m_CurrentServerInstance->state() == QProcess::Running)
+    if( m_CurrentServerInstance && m_CurrentServerInstance->state() == QProcess::Running )
     {
-        LogInfo("Server process started successfully");
-        return true;
+        SetState(Starting);
     }
     else
     {
-        SetLastErrorMessage("Failed to start server process");
-        LogError("Failed to start server process");
+        LogMessage("Failed to start server process");
         return false;
     }
+
+    // Wait for the server to be listening or fail
+    QThread * thread = QThread::currentThread();
+    QTime timeWaited;
+    timeWaited.start();
+    while( timeWaited.elapsed() < 20000 && m_state == Starting )
+    {
+        thread->msleep( 100 );
+        QApplication::processEvents();
+    }
+
+    return GetState() == Running;
```

Comment:
In the case of a timeout, should we set the server m_state to Timeout (or something similar) before returning?

Reply:
Good question. I had this in the previous version but removed it. Now I'm thinking we should probably just kill the process if we have a timeout. We can report it in the log window.

Reply:
Indeed, killing the process sounds cleaner.